src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -134,11 +134,11 @@ public function addCollection(self $collection)
     /**
      * Adds a prefix to the path of all child routes.
      *
-     * @param string $prefix       An optional prefix to add before each pattern of the route collection
-     * @param array  $defaults     An array of default values
-     * @param array  $requirements An array of requirements
+     * @param string|null $prefix       An optional prefix to add before each pattern of the route collection
+     * @param array       $defaults     An array of default values
+     * @param array       $requirements An array of requirements
      */
-    public function addPrefix($prefix, array $defaults = [], array $requirements = [])
+    public function addPrefix(?string $prefix, array $defaults = [], array $requirements = [])
     {
         $prefix = trim(trim($prefix), '/');
```

Comment:
if prefix really needs to be nullable, then we shouldn't call trim on null. this would fail as soon as we use strict types.

Reply:
If we ever were to enable strict types, there are unit tests in place that would fail here. I would not worry about this right now.